### PR TITLE
feat: add selected dropdown element when clicked

### PR DESCRIPTION
**Pull request by whitneypurdum**
- For New Role --> Set Restrictions --> Role Precondition - when an item is selected from the dropdown, automatically add it, rather than relying on the "add" button

### CODE REVIEW

**Comment by whitneypurdum** [comment]
@dawidgil if this implementation is okay, I will update unit tests. I added the condition because this component is shared across the app, so didn't want to force the behavior adoption of adding when dropdown selected.

File: src/app/shared/components/smart-search/smart-search.component.ts [hunk 1]
```diff
@@ -62,8 +63,14 @@ export class SmartSearchComponent implements AfterViewInit {
 
   autocompleteSelectionHandler(event: MatAutocompleteSelectedEvent) {
     this.selectionHandler(event.option.value);
+    if (this.addOnSelect) {
+      this.add.emit({
```
Comment (dawidgil) [inline src/app/shared/components/smart-search/smart-search.component.ts:67]: Here you can reuse `addRole` method. It's the same code, of course inside if statement.

File: src/app/shared/components/smart-search/smart-search.component.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ import { IOrganization } from 'iam-client-lib/dist/src/modules/domains/domains.t
 export class SmartSearchComponent implements AfterViewInit {
   @Input() searchText: FormControl = new FormControl('');
   @Input() placeholderSearch: string;
+  @Input() addOnSelect: boolean;
```
Comment (dawidgil) [inline src/app/shared/components/smart-search/smart-search.component.ts:29]: I suppose you do not need to add this property. You can use `isAdding` getter. It's checking already if it is for adding role or not. However i see that those names do not describes what they are doing. What do you think about renaming them to be more descriptive?
I see two options:
1. Rename `add -> addRole`
2. change `add` interface, by changing `role` into `value`. Method `addRole -> addSelection` This way we can `add` organization/application by using smart search. 
What do you think?
btw. what do you think about removing `add` button from html?

Comment (whitneypurdum) [reply src/app/shared/components/smart-search/smart-search.component.ts:29]: @dawidgil it looks like above it says to change add to addRole, then change addRole to addSelection. Can you clarify? I'm up for removing the add button if everyone is okay with it!

Comment (whitneypurdum) [reply src/app/shared/components/smart-search/smart-search.component.ts:29]: @dawidgil Oh NM, I totally see what you mean!

**Review by dawidgil — APPROVED** [review]
Looks good, thanks!

**Pull request merged**